AudioSinkExample/ViewController.swift

```diff
@@ -0,0 +1,692 @@
+                                                                                  identifier: publication.trackSid)
+        }
+    }
+
+    func unsubscribed(from audioTrack: TVIRemoteAudioTrack,
+                      publication: TVIRemoteAudioTrackPublication,
+                      for participant: TVIRemoteParticipant) {
+
+        // We are unsubscribed from the remote Participant's audio Track. We will no longer receive the
+        // remote Participant's audio.
+
+        logMessage(messageText: "Unsubscribed from \(publication.trackName) audio track for Participant \(participant.identity)")
+
+        if let recorder = self.audioRecorders[publication.trackSid] {
+            recorder.stopRecording()
+            self.audioRecorders.removeValue(forKey: publication.trackSid)
+        }
+
+        if (self.speechRecognizer?.identifier == publication.trackSid) {
+            self.speechRecognizer?.stopRecognizing()
+            self.speechRecognizer = nil
+        }
+    }
+
+    func remoteParticipant(_ participant: TVIRemoteParticipant,
+                           enabledVideoTrack publication: TVIRemoteVideoTrackPublication) {
+        logMessage(messageText: "Participant \(participant.identity) enabled \(publication.trackName) video track")
+    }
+
+    func remoteParticipant(_ participant: TVIRemoteParticipant,
+                           disabledVideoTrack publication: TVIRemoteVideoTrackPublication) {
+        logMessage(messageText: "Participant \(participant.identity) disabled \(publication.trackName) video track")
+    }
+
+    func remoteParticipant(_ participant: TVIRemoteParticipant,
+                           enabledAudioTrack publication: TVIRemoteAudioTrackPublication) {
+        logMessage(messageText: "Participant \(participant.identity) enabled \(publication.trackName) audio track")
+    }
+
+    func remoteParticipant(_ participant: TVIRemoteParticipant,
+                           disabledAudioTrack publication: TVIRemoteAudioTrackPublication) {
+        // We will continue to record silence and/or recognize audio while a Track is disabled.
+        logMessage(messageText: "Participant \(participant.identity) disabled \(publication.trackName) audio track")
+    }
+}
+
+// MARK: TVILocalParticipantDelegate
+extension ViewController : TVILocalParticipantDelegate {
+    func localParticipant(_ participant: TVILocalParticipant, publishedAudioTrack: TVILocalAudioTrackPublication) {
+        // We expect to publish our AudioTrack at Room connect time, but handle a late publish just to be sure.
+        if (recordAudio) {
+            let trackSid = publishedAudioTrack.trackSid
+            self.audioRecorders[trackSid] = ExampleAudioRecorder.init(audioTrack: publishedAudioTrack.localTrack!,
+                                                                      identifier: trackSid)
+            logMessage(messageText: "Recording local audio...")
+        }
```

Comment:
if we are demonstrating this, we need to demonstrate how to cleaning up the recoder from Dictationay in a live call - I am realizing its not easy with our current set of APIs.

Reply:
I'm just handling the case where publish *might* be async, which actually won't happen with our current clients / infra. The recorder is stopped and cleaned up safely when disconnecting from a Room.